pkg/install/resource/tm_deployment.go

```diff
@@ -43,12 +43,13 @@ func (ri *tmDeployment) desiredDeployment(ctx context.Context) *kates.Deployment
 	replicas := int32(1)
 
 	sc := getScope(ctx)
+	imgConfig := client.GetConfig(ctx).Images
 	var containerEnv = []corev1.EnvVar{
 		{Name: "LOG_LEVEL", Value: "info"},
 		{Name: "SYSTEMA_HOST", Value: sc.env.SystemAHost},
 		{Name: "SYSTEMA_PORT", Value: sc.env.SystemAPort},
 		{Name: "CLUSTER_ID", Value: sc.clusterID},
-		{Name: "TELEPRESENCE_REGISTRY", Value: sc.env.Registry},
+		{Name: "TELEPRESENCE_REGISTRY", Value: imgConfig.Registry},
```

Comment:
This (and the agent image further down) is somewhat problematic from a life-cycle perspective. If the manager is installed and a user's config is modified, then that modification will have no effect because it doesn't cause the manager deployment to change. I know that this problem was already there when modifying the env vars but that was not something that we made apparent to the user the way we do now.

I don't have a good solution for this but I think it's important that there's a caveat and an explanation in the docs.

Reply:
I think I have tackled this in a future commit by separating the agentImage key into agentImage and webhookAgentImage so it's not one overloaded variable that is easier to distinguish between in the documentation.

Although revisiting this now, I might need to break up imgConfig.Registry into two variables as well.

Anyways, let me know what you think of this solution in the later commits.